MekHQ/src/mekhq/gui/panes/campaignOptions/tabs/CampaignOptionsUtilities.java

```diff
@@ -0,0 +1,368 @@
+     * title, will be {@code "lbl" + borderTitle + ".text"}
+     *
+     * @param name         the name of the panel.
+     * @param includeBorder whether the panel should have a border.
+     * @param borderTitle  The resource string that should be used as the title of the border.
+     *                    Can be empty to generate an untitled border.
+     *
+     * @return a {@link JPanel} with a titled border and {@link GroupLayout} as its layout manager
+     */
+    static JPanel createStandardPanel(String name, boolean includeBorder, String borderTitle) {
+        borderTitle = borderTitle.isBlank() ? "" : resources.getString("lbl" + borderTitle + ".text");
+
+        JPanel panel = new JPanel();
+
+        if (includeBorder) {
+            panel.setBorder(BorderFactory.createTitledBorder(
+                String.format(String.format("<html><b>%s</b></html>", borderTitle))));
+        }
+
+        panel.setName(name);
+
+        return panel;
+    }
+
+    /**
+     * Creates a {@link JPanel} consisting of a {@link JLabel} above an image.
+     * If {@code includeBodyText} is {@code true} a second {@link JLabel} is placed after the first.
+     * <p>
+     * The {@link JPanel} will be named {@code "pnl" + name + "HeaderPanel"}.
+     * The resource bundle references for the first {@link JLabel} will be {@code "lbl" + name + ".text"}.
+     * The optional second {@link JLabel} is assigned the name {@code ""lbl" + name + "Body"}
+     * and uses the following resource bundle reference: {@code "lbl" + name + "Body.text"}.
+     *
+     * @param name           the name of the header panel.
+     * @param imageAddress   the file path of the image to be displayed in the panel
+     * @param includeBorder  whether the panel should have a border
+     * @param borderTitle    the title of the border; can be empty for an untitled border
+     * @param includeBodyText    if {@code true}, include a second {@link JLabel}.
+     * @return a {@link JPanel} representing the header panel
+     */
+    static JPanel createHeaderPanel(String name, String imageAddress, boolean includeBorder,
+                                           String borderTitle, boolean includeBodyText) {
+        ImageIcon imageIcon = new ImageIcon(imageAddress);
+        JLabel imageLabel = new JLabel(imageIcon);
+
+        final JLabel lblHeader = new JLabel(String.format("<html>%s</html>",
+            resources.getString("lbl" + name + ".text")), SwingConstants.CENTER);
+        lblHeader.setName("lbl" + name);
+
+        JLabel lblBody = new JLabel();
+        if (includeBodyText) {
+            lblBody = new JLabel(String.format("<html>%s</html>",
+                resources.getString("lbl" + name + "Body.text")), SwingConstants.CENTER);
+            lblBody.setName("lbl" + name + "Body");
+            Dimension size = lblBody.getPreferredSize();
+            lblBody.setMaximumSize(new Dimension(500, size.height));
```

Comment:
as above, rather override getPreferredSize() and or max and min. If you use a fixed value, ideally scale it using UIUtil.scaleForGUI(500). There is also the relatively neat option of creating a specialized JLabel that will internally use a secondary JLabel just to size itself. Check out SummaryWeightLabel (in MML). This avoids setting any pixel count and also deals with scaling and even different fonts.

Reply:
Speaking of SummaryWeightLabel, I might have to update it to deal with FlatLaf styling

Reply:
override like this:
... getPreferredSize() { return new Dimension(500, super.getPreferredSize().height); }